x-pack/plugins/task_manager/server/task_scheduling.ts

```diff
@@ -96,11 +110,48 @@ export class TaskScheduling {
    */
   public async runNow(taskId: string): Promise<RunNowResult> {
     return new Promise(async (resolve, reject) => {
-      this.awaitTaskRunResult(taskId).then(resolve).catch(reject);
+      this.awaitTaskRunResult(taskId)
+        // don't expose state on runNow
+        .then(({ id }) => resolve({ id }))
+        .catch(reject);
       this.taskPollingLifecycle.attemptToRun(taskId);
     });
   }
 
+  /**
+   * Run an ad-hoc task in memory without persisting it into ES or distributing the load across the cluster.
+   *
+   * @param task - The ephemeral task being queued.
+   * @returns {Promise<ConcreteTaskInstance>}
+   */
+  public async ephemeralRunNow(
+    tasks: EphemeralTask[],
+    options?: Record<string, unknown>
+  ): Promise<Array<PromiseSettledResult<RunNowResult>>> {
+    return await Promise.allSettled(
+      tasks.map(
+        async (task): Promise<RunNowResult> => {
+          const id = uuid.v4();
+          const { taskInstance: modifiedTask } = await this.middleware.beforeSave({
+            ...options,
+            taskInstance: task,
+          });
+          return new Promise(async (resolve, reject) => {
+            this.awaitTaskRunResult(id).then(resolve).catch(reject);
+            this.ephemeralTaskLifecycle.attemptToRun({
+              id,
+              scheduledAt: new Date(),
+              runAt: new Date(),
+              status: TaskStatus.Idle,
+              ownerId: this.taskManagerId,
+              ...modifiedTask,
+            });
```

Comment:
If this returns an `Err` rather than an `Ok` then you know to call `reject` as the bounded queue was exceeded.

Reply:
Ahhhh looking further through the code I think I get it - you use `isEphemeralTaskDelayedDueToCapacityEvent` to detect that case and rerject.
The problem is that the `EphemeralTaskDelayedDueToCapacityEvent` is created when the TaskPool has no capacity which means that if there are 10 Rules running, and one schedules a single action - that action will be rejected, instead of queued ephemerally.
So our effective ephemeral queue is `MAX_WORKERS - current running tasks` which is likely to be a very small number (if not 0) much of the time.